src/server/src/main/java/io/cassandrareaper/management/http/HttpMetricsProxy.java

```diff
@@ -17,49 +17,211 @@
 
 package io.cassandrareaper.management.http;
 
+import io.cassandrareaper.ReaperException;
 import io.cassandrareaper.core.GenericMetric;
+import io.cassandrareaper.core.JmxStat;
 import io.cassandrareaper.core.Node;
 import io.cassandrareaper.management.MetricsProxy;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
 import javax.management.JMException;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
 public final class HttpMetricsProxy implements MetricsProxy {
 
+  private static final String THREAD_POOL_METRICS_PREFIX = "org_apache_cassandra_metrics_thread_pools";
+  private static final String TPSTATS_PENDING_METRIC_NAME = "org_apache_cassandra_metrics_thread_pools_pending_tasks";
   private HttpCassandraManagementProxy proxy;
+  private final OkHttpClient httpClient;
+  private final Node node;
 
-  private HttpMetricsProxy(HttpCassandraManagementProxy proxy) {
+  private HttpMetricsProxy(HttpCassandraManagementProxy proxy,
+                           Node node,
+                           Supplier<OkHttpClient> httpClientSupplier) {
     this.proxy = proxy;
+    this.node = node;
+    this.httpClient = httpClientSupplier.get();
+  }
+
+  public static HttpMetricsProxy create(HttpCassandraManagementProxy proxy, Node node) {
+    return new HttpMetricsProxy(proxy, node, () -> new OkHttpClient.Builder().build());
   }
 
-  public static HttpMetricsProxy create(HttpCassandraManagementProxy proxy) {
-    return new HttpMetricsProxy(proxy);
+  @VisibleForTesting
+  static HttpMetricsProxy create(
+      HttpCassandraManagementProxy proxy, Node node, Supplier<OkHttpClient> httpClientSupplier) {
+    return new HttpMetricsProxy(proxy, node, httpClientSupplier);
   }
 
   @Override
-  public List<GenericMetric> collectTpStats(Node node) throws JMException, IOException {
-    throw new UnsupportedOperationException("Not implemented");
+  public List<GenericMetric> collectTpStats() throws JMException, IOException {
+    // Collect all metrics and filter out the ones that are not thread pool metrics
+    List<GenericMetric> genericMetrics = MetricsProxy.convertToGenericMetrics(collectMetrics(), node);
```

Comment:
issue: This isn't calling ``collectMetrics`` with correct parameters. Should be ``collectMetrics(Optional.of(THREAD_POOL_METRICS_PREFIX))``.